plugin/clang.vim

```diff
@@ -840,10 +840,27 @@ func! s:ClangCompleteInit(force)
 endf
 "}}}
 "{{{ ClangExecuteNeoJobHandler
-"handles event: exit
+"Handles events: stdout, stderr, exit
 func! ClangExecuteNeoJobHandler(job_id, data, event)
-  if a:event == 'exit'
-    call ClangExecuteDone(self.fstdout, self.fstderr)
+  if a:event == 'stdout' || a:event == 'stderr'
+    if len(b:clang_state[a:event]) != 0
+      if b:clang_state[a:event][-1] == ''
+        call remove(b:clang_state[a:event], -1)
+      else
+        let line = a:data[0]
+        let b:clang_state[a:event][-1] .= line
+        call remove(a:data, 0)
+      endif
+    endif
```

Comment:
:smile:
There's no mention of this in the poor documentation... So it's hard to understand...Comment is needed when have some complex logic.
```vim
if a:event == 'stdout' || a:event == 'stderr'
  " when a:data[-1] is empty, which means is a complete line, otherwise need to concat a:data[-1]
  if !empty(b:clang_state[a:event])
    if empty(b:clang_state[a:event][-1])
      " a complete line, just remove the last empty line
      call remove(b:clang_state[a:event], -1)
    else
      " need to concat to the last line in previous chunk
      let b:clang_state[a:event][-1] .= a:data[0]
      call remove(a:data, 0)
    endif
  endif
  let b:clang_state[a:event] += a:data
else
```